html/includes/authentication/sso.inc.php

```diff
@@ -0,0 +1,336 @@
+<?php
+/**
+ * sso.inc.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2017 Adam Bishop
+ * @author     Adam Bishop <adam@omega.org.uk>
+ */
+
+use LibreNMS\Config;
+use LibreNMS\Util\IP;
+use LibreNMS\Exceptions\AuthenticationException;
+use LibreNMS\Exceptions\InvalidIpException;
+
+/**
+ * Some functionality in this mechanism is inspired by confluence_http_authenticator (@chauth) and graylog-plugin-auth-sso (@Graylog)
+ *
+ */
+
+function init_auth()
+{
+    // No initialisation is required
+}
+
+function authenticate($username, $password)
+{
+    if (empty($username)) {
+        throw new AuthenticationException('sso.user_attr was not found or was empty');
```

Comment:
Users don't see config values like that so you will need to `$config['sso']['user_attr']` in this output

Reply:
Done.